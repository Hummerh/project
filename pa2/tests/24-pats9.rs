fn main() {
      let a : [bool;3] = [true, true, true];
      let [mut x, y, z] = a;

      let [[mut m], [mut n], [mut o], [mut p]] = [[1], [2], [3], [4]];

      match ([[1],[2],[3],[4]]) {
            [_, b, c, _] => {m + n + o + p;}
      };

      match ([[1]]) {
            [[2]] => { },
            [[x]] => { }
      };
}
