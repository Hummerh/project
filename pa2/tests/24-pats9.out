(crate
    (items
        (fn-def
            (id
                (main))
            (block
                (let
                    (pat-id
                        (id
                            (a)))
                    (type-arr
                        (type-bool)
                        (lit-dec))
                    (arr
                        (exprs
                            (true)
                            (true)
                            (true))))
                (let
                    (pat-arr
                        (pat-arr-elems
                            (pat-mut-id
                                (id
                                    (x)))
                            (pat-id
                                (id
                                    (y)))
                            (pat-id
                                (id
                                    (z)))))
                    (id
                        (a)))
                (let
                    (pat-arr
                        (pat-arr-elems
                            (pat-arr
                                (pat-arr-elems
                                    (pat-mut-id
                                        (id
                                            (m)))))
                            (pat-arr
                                (pat-arr-elems
                                    (pat-mut-id
                                        (id
                                            (n)))))
                            (pat-arr
                                (pat-arr-elems
                                    (pat-mut-id
                                        (id
                                            (o)))))
                            (pat-arr
                                (pat-arr-elems
                                    (pat-mut-id
                                        (id
                                            (p)))))))
                    (arr
                        (exprs
                            (arr
                                (exprs
                                    (lit-dec)))
                            (arr
                                (exprs
                                    (lit-dec)))
                            (arr
                                (exprs
                                    (lit-dec)))
                            (arr
                                (exprs
                                    (lit-dec))))))
                (match
                    (arr
                        (exprs
                            (arr
                                (exprs
                                    (lit-dec)))
                            (arr
                                (exprs
                                    (lit-dec)))
                            (arr
                                (exprs
                                    (lit-dec)))
                            (arr
                                (exprs
                                    (lit-dec)))))
                    (match-arms
                        (match-arm
                            (pats
                                (pat-arr
                                    (pat-arr-elems
                                        (pat-wild)
                                        (pat-id
                                            (id
                                                (b)))
                                        (pat-id
                                            (id
                                                (c)))
                                        (pat-wild))))
                            (block
                                (add
                                    (add
                                        (add
                                            (id
                                                (m))
                                            (id
                                                (n)))
                                        (id
                                            (o)))
                                    (id
                                        (p)))))))
                (match
                    (arr
                        (exprs
                            (arr
                                (exprs
                                    (lit-dec)))))
                    (match-arms
                        (match-arm
                            (pats
                                (pat-arr
                                    (pat-arr-elems
                                        (pat-arr
                                            (pat-arr-elems
                                                (pat-lit
                                                    (lit-dec)))))))
                            (block))
                        (match-arm
                            (pats
                                (pat-arr
                                    (pat-arr-elems
                                        (pat-arr
                                            (pat-arr-elems
                                                (pat-id
                                                    (id
                                                        (x))))))))
                            (block))))))))